AndroidRepository/Extensions.cs

```diff
@@ -0,0 +1,219 @@
+﻿using AndroidRepository.Addon2_3;
+using AndroidRepository.AddonsList_6;
+using AndroidRepository.Common_2;
+using AndroidRepository.SitesCommon_1;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using static System.Net.Mime.MediaTypeNames;
+
+namespace AndroidRepository;
+
+public static class ChannelTypes
+{
+	public const string Stable = "stable";
+	public const string Beta = "beta";
+	public const string Dev = "dev";
+	public const string Canary = "canary";
+	public static string GetBaseUrl(this RepositoryType repository)
+	{
+		if (repository == null)
+			throw new ArgumentNullException(nameof(repository));
+
+		if (string.IsNullOrEmpty(repository.Url))
+			throw new ArgumentException("Repository URL cannot be null or empty.", nameof(repository));
+
+		var uri = new Uri(repository.Url);
+		var baseUrl = uri.GetLeftPart(UriPartial.Path).TrimEnd('/');
+		return baseUrl;
+	}
+
+}
+
+public static class Extensions
+{
+	public static string GetLicenseFileHash(this Common_2.LicenseType licenseType)
+	{
```

Comment:
Snuck this in semi on accident early... But this will calculate the actual license hash that goes into the license directory and license files that the SDK checks to see if you've accepted... Will eventually add more work in another PR to leverage this for checking license state on individual licenses more easily.